cluster-autoscaler/core/scale_down.go

```diff
@@ -336,18 +361,27 @@ func (sd *ScaleDown) TryToScaleDown(nodes []*apiv1.Node, pods []*apiv1.Pod, pdbs
 	// Nothing super-bad should happen if the node is removed from tracker prematurely.
 	simulator.RemoveNodeFromTracker(sd.usageTracker, toRemove.Node.Name, sd.unneededNodes)
 	nodeDeletionStart := time.Now()
-	err = deleteNode(sd.context, toRemove.Node, toRemove.PodsToReschedule)
+
+	// Starting deletion.
 	nodeDeletionDuration = time.Now().Sub(nodeDeletionStart)
-	if err != nil {
-		return ScaleDownError, err.AddPrefix("Failed to delete %s: ", toRemove.Node.Name)
-	}
-	if readinessMap[toRemove.Node.Name] {
-		metrics.RegisterScaleDown(1, metrics.Underutilized)
-	} else {
-		metrics.RegisterScaleDown(1, metrics.Unready)
-	}
+	sd.nodeDeletionStatus.SetDeleteInProgress(true)
+	go func() {
+		// Finishing the delete probess once this goroutine is over.
+		defer sd.nodeDeletionStatus.SetDeleteInProgress(false)
+
+		err = deleteNode(sd.context, toRemove.Node, toRemove.PodsToReschedule)
+		if err != nil {
+			glog.Errorf("Failed to delete %s: %v", toRemove.Node.Name, err)
+			return
```

Comment:
We should bump up errors metric in here (previously we would return error which would cause that to happen later on).

Reply:
Next PR.